src/utils/export/index.ts

```diff
@@ -77,3 +93,70 @@ export const getGoogleLikeMetadataFile = (
         2
     );
 };
+
+export const sanitizeName = (name: string) =>
+    name.replaceAll('/', '_').replaceAll(' ', '_');
+
+export const getUniqueCollectionFolderPath = (
+    dir: string,
+    collectionName: string,
+    usedCollectionPaths: Set<string>
+): string => {
+    let collectionFolderPath = `${dir}/${sanitizeName(collectionName)}`;
+    let count = 1;
+    while (usedCollectionPaths.has(collectionFolderPath)) {
+        collectionFolderPath = `${dir}/${sanitizeName(
+            collectionName
+        )}(${count})`;
+        count++;
+    }
+    usedCollectionPaths.add(collectionFolderPath);
+    return collectionFolderPath;
+};
+
+export const getMetadataFolderPath = (collectionFolderPath: string) =>
+    `${collectionFolderPath}/${METADATA_FOLDER_NAME}`;
+
+export const getUniqueFileSaveName = (
+    filename: string,
+    usedFilenamesInCollection: Set<string>
+) => {
+    let fileSaveName = sanitizeName(filename);
+    const count = 1;
+    while (usedFilenamesInCollection.has(fileSaveName)) {
+        const filenameParts = splitFilenameAndExtension(fileSaveName);
+        if (filenameParts[1]) {
+            fileSaveName = `${filenameParts[0]}(${count}).${filenameParts[1]}`;
+        } else {
+            fileSaveName = `${filenameParts[0]}(${count})`;
+        }
+    }
+    usedFilenamesInCollection.add(fileSaveName);
+    return fileSaveName;
+};
+
+export const getFileMetadataSavePath = (
+    collectionFolderPath: string,
+    fileSaveName: string
+) => `${collectionFolderPath}/${METADATA_FOLDER_NAME}/${fileSaveName}.json`;
```

Comment:
why use filename for metadata instead of file_id? What will be the behaviour when two files have same file name?

Reply:
just named them as google photos names them....by appending `.json` to the filename the metadata is associated with.

the fileSaveName is the name after appending numbered suffixes , and so it takes care of multiple file with same name issue.

Reply:
Okay. As our export is incremental, what would be the behaviour if a file is exported and the user decides to rename the file?

Reply:
we would re-export the the file as the file's updatedAt time will change
and when we check for files where are uploaded after last upload [1]. we would consider it as one because the entry for a exported file includes the `updationTime` [2], so we would no entry for this file

[1] https://github.com/ente-io/bada-frame/blob/export-v2/src/utils/export/index.ts#L23
[2] https://github.com/ente-io/bada-frame/blob/export-v2/src/utils/export/index.ts#L7